server.js

```diff
@@ -35,14 +37,26 @@ app.use(express.static('.', {
     }
 }));
 
-// // endpoint for updating the app
-// app.post('/git', (req, res) => {
-//     if (req.headers['x-github-event'] == "push") { 
-//         /* Here will be our updating code */
-//     }
-
-//     return res.sendStatus(200);
-// });
+// endpoint for updating the app (https://support.glitch.com/t/tutorial-how-to-auto-update-your-project-with-github/8124)
+app.post('/git', (req, res) => {
+    const hmac = crypto.createHmac('sha1', process.env.SECRET);
+    const sig  = 'sha1=' + hmac.update(JSON.stringify(req.body)).digest('hex');
+
+    if (req.headers['x-github-event'] == 'push' && sig == req.headers['x-hub-signature']) { 
```

Comment:
It doesn't matter for this endpoint currently, but the '==' string comparison is vulnerable to a timing attack, so it's possible for someone to craft a body and find the right signature that would pass without too much effort. To prevent this, we can use [crypto.timingSafeEqual](https://nodejs.org/api/crypto.html#crypto_crypto_timingsafeequal_a_b) instead of == for the signature comparison.

That way if we ever do anything sensitive here, it won't be vulnerable.

Reply:
TIL! I had no idea. Thank you for bringing this up.